### ISSUE TITLE
Overwrite configuration is not taken into account

### ISSUE THREAD

**Issue by sruegg** [einpraegsam/powermail#2]
Hi,

Just found an issue when sender.name and sender.email are not set (TypoScript or Flexform).
I've got an overwrite configuration (and works fine).

The problems is that replyToName and replyToEmail are empty when creating the senderEmail. Specially in \In2code\Powermail\Domain\Service\SendMailService::prepareAndSend. The call to setReplyTo trigger error as empty strings aren't valid.

Email is not sent at the end!

I don't really know if this is wanted or not?

P.S.: setting sender.name and sender.email with dummy (finally unused) values fix the whole thing.

**Comment by einpraegsam** [comment]
Can you please send me a screenshot of your FlexForm settings? And in addition I need your TypoScript.

**Comment by sruegg** [comment]
Here is my TS constants:
```
plugin.tx_powermail {
    view {
        layoutRootPath = EXT:my_skin/Resources/Private/Layouts/Powermail/
        partialRootPath = EXT:my_skin/Resources/Private/Partials/Powermail/
        templateRootPath = EXT:my_skin/Resources/Private/Templates/Powermail/
    }

    settings {
        receiver {
            default {
                senderName = My sender name
                senderEmail = email@domain.com
            }
        }

        sender {
            default {
                senderName = My sender name
                senderEmail = email@domain.com
            }
        }
    }
}
```
We are getting here some information from database to set later (overwrite) receiver.
My TS setup (but even with no custom configuration at all emails are not sent):
```
# Récupération de l'uid de l'utilisateur FE
lib.useruid = CONTENT
lib.useruid {
    table = tx_tccontactme_domain_model_contactsubject
    select {
        pidInList = {$plugin.tx_tccontactme.persistence.storagePid}
        andWhere {
            data = GP:tx_powermail_pi1|field|subject
            wrap = tx_tccontactme_domain_model_contactsubject.uid=|
            intval = 1
        }
    }
    renderObj = TEXT
    renderObj {
        stdWrap.field = person
    }
}

# Récupération de l'email de l'utilisateur FE
lib.pmreceiver = CONTENT
lib.pmreceiver {
    table = fe_users
    select {
        pidInList = {$plugin.tx_tccontactme.persistence.storagePid}
        andWhere {
            stdWrap.postCObject < lib.useruid
            wrap = fe_users.uid=
            intval = 1
        }
    }

    renderObj = TEXT
    renderObj {
        stdWrap.field = email
    }
}

# Récupération du nom de l'utilisateur FE
lib.pmreceiverName = CONTENT
lib.pmreceiverName {
    table = fe_users
    select {
        pidInList = {$plugin.tx_tccontactme.persistence.storagePid}
        andWhere {
            stdWrap.postCObject < lib.useruid
            wrap = fe_users.uid=
            intval = 1
        }
    }

    renderObj = COA
    renderObj {
        10 = TEXT
        10.stdWrap.field = first_name

        20 = TEXT
        20.noTrimWrap = | ||

        30 = TEXT
        30.stdWrap.field = last_name
    }
}

# Récupération de l'email de l'utilisateur
lib.pmsenderEmail = TEXT
lib.pmsenderEmail {
    data = GP:tx_powermail_pi1|field|email
}

# Récupération du prénom/nom de l'utilisateur
lib.pmsenderName = COA
lib.pmsenderName {
    10 = TEXT
    10.data = GP:tx_powermail_pi1|field|firstname

    20 = TEXT
    20.noTrimWrap = | ||

    30 = TEXT
    30.data = GP:tx_powermail_pi1|field|lastname
}

# Génération du sujet de l'email
lib.pmsubject = CONTENT
lib.pmsubject {
    table = tx_tccontactme_domain_model_contactsubject
    select {
        pidInList = {$plugin.tx_tccontactme.persistence.storagePid}
        andWhere {
            data = GP:tx_powermail_pi1|field|subject
            wrap = tx_tccontactme_domain_model_contactsubject.uid=|
            intval = 1
        }
    }

    renderObj = TEXT
    renderObj {
        stdWrap.field = name
        stdWrap.noTrimWrap = |Amjet Executive - Demande de contact: ||
    }
}

plugin.tx_powermail {
    settings {
        setup {
            spamshield._enable = 0

            # Configurations admin
            receiver {
                overwrite {
                    subject < lib.pmsubject
                    email < lib.pmreceiver

                    replyToName < lib.pmsenderName
                    replyToEmail < lib.pmsenderEmail

                    senderName.value = {$plugin.tx_powermail.settings.receiver.default.senderName}
                    senderEmail.value = {$plugin.tx_powermail.settings.receiver.default.senderEmail}

                    name < lib.pmreceiverName
                }
            }

            # Configuration utilisateur
            sender {
                enable = 1
#################### HERE #################
                email = {$plugin.tx_powermail.settings.sender.default.senderEmail}
                name = {$plugin.tx_powermail.settings.sender.default.senderName}
####### errors comes if those 2 are empty #########
                overwrite {
                    subject < lib.pmsubject
                    email < lib.pmsenderEmail
                    replyToName.value = {$plugin.tx_powermail.settings.receiver.overwrite.name}
                    replyToEmail < lib.pmreceiver
                    senderName.value = {$plugin.tx_powermail.settings.receiver.default.senderName}
                    senderEmail.value = {$plugin.tx_powermail.settings.receiver.default.senderEmail}
                }
            }

            excludeFromPowermailAllMarker {
                confirmationPage {
                    excludeFromMarkerNames = receiver, subject
                }

                submitPage < .confirmationPage
                receiverMail < .confirmationPage
                senderMail < .confirmationPage
                optinMail < .confirmationPage
            }
        }
    }
}
```

And screenshots of my Flexform settings (nothing really set here):
![image](https://cloud.githubusercontent.com/assets/777457/19683384/556c11c0-9ab2-11e6-8f3e-56138fd39b0b.png)
![image](https://cloud.githubusercontent.com/assets/777457/19683407/5c3f5354-9ab2-11e6-81e9-bba642c342ff.png)

**Comment by marceltoepler** [comment]
I can confirm that this functionality stopped working in version 3.8.0.
My ts setup looks like:
```
plugin.tx_powermail.settings.setup {
    receiver.overwrite {
        senderEmail = TEXT
        senderEmail.value = no-reply@xxx.de

        senderName = TEXT
        senderName.value = xxx
    }
}
```
In version 3.7.0 this works fine but it seems like it stopped after update to 3.8.0.

**Comment by einpraegsam** [comment]
I'm a bit confused now. So this "In version 3.7.0 this works fine but it seems like it stopped after update to 3.8.0" is wrong?

**Comment by sruegg** [comment]
For me my configuration seems to stop working after version > 3.1.1. But as I said, setting sender.name and sender.email to dummy values make it work again. Receiver part seems not impacted by this (as I know).

**Comment by marceltoepler** [comment]
Yes, you can completely ignore my comment.

**Comment by einpraegsam** [comment]
Sorry, I still don't get it while I was playing with this issue. And it's not that easy because of the configuratino and the relation between it.
If I turn off all FlexForm and TypoScript Settings of the Sender Email - this works like a charm:
```
plugin.tx_powermail.settings.setup.receiver.default {
  senderName = TEXT
  senderName.value = My sender name
  senderEmail = TEXT
  senderEmail.value = email@domain.com
}
```
Of course also the checkbox "Sender Email" must not be checked in form settings.

But this leads to the situation that no email could be send to the sender, because the checkbox is not set. If you want also such an email, you have to check the checkbox (than this email-address is taken also for sender email to receiver) or you are using the overwrite settings in TypoScript (not the default settings)

Long story short: I can't find an issue

**Comment by cweiske** [comment]
@einpraegsam: I experience the same issue. Sender email ("Mail to user") does not get sent, the form shows an error:

> Failure, mail could not be sent!

This happens under following circumstances:
- Plugin settings -> Mail to User: 
 - Sender Name + Sender Email are empty
 - Subject for Email to sender is filled with some string
- TypoScript:
 - `plugin.tx_powermail.settings.setup.sender.overwrite.senderEmail.value` and `plugin.tx_powermail.settings.setup.sender.overwrite.senderName.value`  are set to some values

When submitting the form now, the error occurs.

When I remove my `.sender.*` TypoScript configuration, the mail to the user does not get sent at all.

**Comment by cweiske** [comment]
@einpraegsam could you reevaluate this bug and reopen it if you can reproduce it?

**Comment by cweiske** [comment]
Seems I get this fixed with my new patch to #74.
